src/libre/ast_compile_internal.h

```diff
@@ -0,0 +1,162 @@
+#ifndef AST_COMPILE_INTERNAL_H
+#define AST_COMPILE_INTERNAL_H
```

Comment:
if this stuff is really only used by ast_compile.c, i'd rather we write it directly in that .c file. likewise for the other *_internal.h headers. having a separate header means i'd need to read this at the same time as the .c file in order to understand the structs and such when they're only ever used there, but also hints that other .c files might include these too, but we're not introducing an internal api here

Reply:
Okay. I specifically moved them into their own files so I could pull up the struct definitions, prototypes, etc. alongside the code in another buffer, not because I expect anything else to call them.

Reply:
Updated in 4b1f511e523054ce4f892dbcf07728217a49ec50.